### PR TITLE
Add indices according to auto recommendations

### PR DESCRIPTION
**Pull request by skofman1**
@maartenba @xavierdecoster @scottbommarito @shishirx34 @ryuyu

### CODE REVIEW

File: src/NuGetGallery/Migrations/201609092255576_AddIndexUsersEmail.cs [hunk 1]
```diff
@@ -0,0 +1,17 @@
+namespace NuGetGallery.Migrations
+{
+    using System.Data.Entity.Migrations;
+    
+    public partial class AddIndexUsersEmail : DbMigration
+    {
+        public override void Up()
+        {
+            Sql("IF NOT EXISTS(SELECT * FROM sys.indexes WHERE name = 'nci_wi_Users_EmailAllowed' AND object_id = OBJECT_ID('Users')) CREATE NONCLUSTERED INDEX [nci_wi_Users_EmailAllowed] ON [dbo].[Users] ([EmailAllowed], [EmailAddress]) INCLUDE ([Key], [Username]) WITH (ONLINE = ON)");
```
Comment (xavierdecoster) [inline src/NuGetGallery/Migrations/201609092255576_AddIndexUsersEmail.cs]: Once applied in PROD, we need to remove the `WITH ONLINE=ON` part of this query to support people running the gallery on non-Enterprise SqlServer versions.

File: src/NuGetGallery/Migrations/201609092252096_AddIndexCredentialExpires.cs [hunk 2]
```diff
@@ -0,0 +1,17 @@
+namespace NuGetGallery.Migrations
+{
+    using System.Data.Entity.Migrations;
+    
+    public partial class AddIndexCredentialExpires : DbMigration
+    {
+        public override void Up()
+        {
+            Sql("IF NOT EXISTS(SELECT * FROM sys.indexes WHERE name = 'nci_wi_Credentials_Type_Expires' AND object_id = OBJECT_ID('Credentials')) CREATE NONCLUSTERED INDEX [nci_wi_Credentials_Type_Expires] ON [dbo].[Credentials] ([Type], [Expires]) INCLUDE ([Created], [UserKey]) WITH (ONLINE = ON)");
```
Comment (xavierdecoster) [inline src/NuGetGallery/Migrations/201609092252096_AddIndexCredentialExpires.cs]: Once applied in PROD, we need to remove the `WITH ONLINE=ON` part of this query to support people running the gallery on non-Enterprise SqlServer versions.

**Comment by xavierdecoster** [comment]
One comment regarding post-deployment. Otherwise LGTM. :ship:

**Comment by maartenba** [comment]
LGTM, do make sure to drop the `WITH ONLINE=ON` part after applying this one to production - it may break private gallery deployments if kept around.

**Pull request merged**